config.c

```diff
@@ -1711,19 +1721,21 @@ static int do_git_config_sequence(const struct config_options *opts,
 	else
 		repo_config = NULL;
 
-	current_parsing_scope = CONFIG_SCOPE_SYSTEM;
-	if (git_config_system() && !access_or_die(git_etc_gitconfig(), R_OK,
-						  opts->system_gently ?
-						  ACCESS_EACCES_OK : 0))
-		ret += git_config_from_file(fn, git_etc_gitconfig(),
-					    data);
+	if(!override) {
+		current_parsing_scope = CONFIG_SCOPE_SYSTEM;
+		if (git_config_system() && !access_or_die(git_etc_gitconfig(), R_OK,
+							  opts->system_gently ?
+							  ACCESS_EACCES_OK : 0))
+			ret += git_config_from_file(fn, git_etc_gitconfig(),
+						    data);
```

Comment:
An alternative would be to turn `GIT_CONFIG_NOSYSTEM` into a multi-state: instead of treating it exclusively as a Boolean, we could also allow a prefix `override:` followed by the path.

In any case, it would look more elegant to me if we patched `git_config_system()` instead of now having _two_ blocks:

https://github.com/git/git/blob/b680c9b7a118c675fa2e621707ccc5f9f5719e80/config.c#L1693-L1697

would become

```c
const char *git_config_system(void)
{
	static int initialized;
	static const char *path;
	const char *env, *p;

	if (initialized)
		return path;

	env = getenv("GIT_CONFIG_NOSYSTEM");
	if (env && skip_prefix(env, "override:", &p))
		path = expand_user_path(p, 0);
	else if (env && git_config_bool("GIT_CONFIG_NOSYSTEM", env))
		path = git_etc_gitconfig();

	initialized = 1;
	return path;
}
```

The `do_git_config_sequence()` function would then have `const char *system = git_config_system();` in the beginning, and would then have once again only one block of code responsible for parsing the system config.